include/unifex/let_value_with_stop_token.hpp

```diff
@@ -0,0 +1,178 @@
+        unifex::set_value(std::move(receiver_), (Values&&)values...);
+    }
+
+    template(typename Error)
+        (requires receiver<Receiver, Error>)
+    void set_error(Error&& error) noexcept {
+        unifex::set_error(std::move(receiver_), (Error&&)error);
+    }
+
+    void set_done() noexcept {
+        unifex::set_done(std::move(receiver_));
+    }
+
+    friend inplace_stop_token tag_invoke(tag_t<get_stop_token>, const type& r) noexcept {
+        return r.op_->stop_token_;
+    }
+
+    template(typename CPO, typename Self)
+        (requires
+            is_receiver_query_cpo_v<CPO> AND
+            same_as<Self, type>)
+    friend auto tag_invoke(CPO cpo, const Self& self)
+        noexcept(is_nothrow_callable_v<CPO, const Receiver&>)
+        -> callable_result_t<CPO, const Receiver&> {
+        return cpo(self.receiver_);
+    }
+
+private:
+    Operation* op_;
+    UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
+};
+
+template<typename SuccessorFactory>
+struct _stop_token_sender {
+    class type;
+};
+
+template<typename SuccessorFactory>
+using stop_token_sender = typename _stop_token_sender<SuccessorFactory>::type;
+
+template<typename SuccessorFactory>
+class _stop_token_sender<SuccessorFactory>::type {
+    template<typename... Values>
+    using additional_arguments = type_list<type_list<Values...>>;
+
+public:
+    using InnerOp = std::invoke_result_t<SuccessorFactory, inplace_stop_token&>;
+
+    template<template<typename...> class Variant, template<typename...> class Tuple>
+    using value_types = sender_value_types_t<InnerOp, Variant, Tuple>;
+
+    template<template<typename...> class Variant>
+    using error_types = sender_error_types_t<InnerOp, Variant>;
+
+    static constexpr bool sends_done = sender_traits<InnerOp>::sends_done;
+
+    template<typename SuccessorFactory2>
+    explicit type(SuccessorFactory2&& func) : func_((SuccessorFactory2&&)func)
+    {}
+
+    template(typename Self, typename Receiver)
+        (requires same_as<remove_cvref_t<Self>, type> AND receiver<Receiver>)
+    friend auto tag_invoke(tag_t<unifex::connect>, Self&& self, Receiver&& r)
+        noexcept(
+            std::is_nothrow_invocable_v<member_t<Self, SuccessorFactory>, inplace_stop_token&> &&
+            std::is_nothrow_constructible_v<remove_cvref_t<Receiver>, Receiver>)
+        -> operation<member_t<Self, SuccessorFactory>, Receiver> {
+        return operation<member_t<Self, SuccessorFactory>, Receiver>(
+            static_cast<Self&&>(self).func_, static_cast<Receiver&&>(r));
+    }
+
+private:
+    UNIFEX_NO_UNIQUE_ADDRESS SuccessorFactory func_;
+};
+
+template<typename SuccessorFactory, typename Receiver>
+struct _stop_token_operation<SuccessorFactory, Receiver>::type {
+
+    template <typename SuccessorFactory2, typename Receiver2>
+    type(SuccessorFactory2&& func, Receiver2&& r) :
+        func_{(SuccessorFactory2&&)func},
+        stop_token_{stop_token_adapter_.subscribe(get_stop_token(r))},
```

Comment:
bug: You are calling `.subscribe()` here but never calling `.unsubscribe()`.

The `inplace_stop_token_adapter` type is like `manual_lifetime` in that it requires you to explicitly unsubscribe to clean it up - it doesn't keep track of whether or not there is a current subscription active or not.

Also, be careful to handle the case where the call to `connect()` below throws.